var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -281,24 +282,26 @@ def filter_compilers(self):
         filter_file(env['CC'],  cc_compiler, nrniv_makefile, **kwargs)
         filter_file(env['CXX'], cxx_compiler, nrniv_makefile, **kwargs)
 
+    @when('+python')
```

Comment:
we always call `set_python_path` from `setup_environment`. What `@when` decorators does here?

Reply:
the decorator changes the original func. So even if we call it directly we call the "no-op" function

Reply:
Ok understood